src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ReferralService.kt

```diff
@@ -1,24 +1,38 @@
 package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.service
 
-import org.springframework.data.repository.findByIdOrNull
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config.Code
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config.FieldError
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config.ValidationError
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.DraftReferralDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralRepository
 import java.time.LocalDate
+import java.time.OffsetDateTime
 import java.util.UUID
 
 @Service
 class ReferralService(val repository: ReferralRepository) {
-  fun createDraftReferral(userID: String, authSource: String): Referral {
-    return repository.save(Referral(createdByUserID = userID, createdByUserAuthSource = authSource))
+  fun getSentReferral(id: UUID): Referral? {
+    return repository.findByIdAndSentAtIsNotNull(id)
+  }
+
+  fun sendDraftReferral(referral: Referral, user: AuthUser): Referral {
+    referral.sentAt = OffsetDateTime.now()
+    referral.sentBy = user
+
+    // fixme: hardcoded for now
+    referral.referenceNumber = "HDJ2123F"
+    return repository.save(referral)
+  }
+
+  fun createDraftReferral(user: AuthUser): Referral {
+    return repository.save(Referral(createdBy = user))
   }
 
   fun getDraftReferral(id: UUID): Referral? {
-    return repository.findByIdOrNull(id)
+    return repository.findByIdAndSentAtIsNull(id)
```

Comment:
_wider question, later_
Returning `null` for already sent referrals is safe but potentially confusing for clients. Getting a 404 on the previously valid request can be confusing.

Is there a way (later) to respond with "303 See Other" or "409 Conflict"?

Reply:
hmm let's discuss outside this pr. it definitely shouldn't be a 200 code IMO, but happy to think about others